src/test/youtube_test.py

```diff
@@ -0,0 +1,127 @@
+print(ax)
+ax = sns.barplot(x='Channel_name', y='Total_videos', data=channel_data)
+print(ax)
+print(channel_data)
+playlist_id = channel_data.loc[channel_data['Channel_name']=='Ken Jee', 'playlist_id'].iloc[0]
+print(playlist_id)
+def get_video_ids(youtube, playlist_id):
+    
+    request = youtube.playlistItems().list(
+                part='contentDetails',
+                playlistId = playlist_id,
+                maxResults = 50)
+    response = request.execute()
+    
+    video_ids = []
+    
+    for i in range(len(response['items'])):
+        video_ids.append(response['items'][i]['contentDetails']['videoId'])
+        
+    next_page_token = response.get('nextPageToken')
+    more_pages = True
+    
+    while more_pages:
+        if next_page_token is None:
+            more_pages = False
+        else:
+            request = youtube.playlistItems().list(
+                        part='contentDetails',
+                        playlistId = playlist_id,
+                        maxResults = 50,
+                        pageToken = next_page_token)
+            response = request.execute()
+    
+            for i in range(len(response['items'])):
+                video_ids.append(response['items'][i]['contentDetails']['videoId'])
+            
+            next_page_token = response.get('nextPageToken')
+        
+    return video_ids
+video_ids = get_video_ids(youtube, playlist_id)
+print(video_ids)
+def get_video_details(youtube, video_ids):
+    all_video_stats = []
+    
+    for i in range(0, len(video_ids), 50):
+        request = youtube.videos().list(
+                    part='snippet,statistics',
+                    id=','.join(video_ids[i:i+50]))
+        response = request.execute()
+        
+        for video in response['items']:
+            video_stats = dict(Title = video['snippet']['title'],
+                               Published_date = video['snippet']['publishedAt'],
+                               Views = video['statistics']['viewCount'],
+                               Likes = video['statistics']['likeCount'],
+                               Dislikes = video['statistics']['dislikeCount'],
+                               Comments = video['statistics']['commentCount']
+                               )
+            all_video_stats.append(video_stats)
+    
+    return all_video_stats
+video_details = get_video_details(youtube, video_ids)
+
+
+video_data = pd.DataFrame(video_details)
+video_data['Published_date'] = pd.to_datetime(video_data['Published_date']).dt.date
+video_data['Views'] = pd.to_numeric(video_data['Views'])
+video_data['Likes'] = pd.to_numeric(video_data['Likes'])
+video_data['Dislikes'] = pd.to_numeric(video_data['Dislikes'])
+video_data['Views'] = pd.to_numeric(video_data['Views'])
+print(video_data)
+top10_videos = video_data.sort_values(by='Views', ascending=False).head(10)
+print(top10_videos)
+ax1 = sns.barplot(x='Views', y='Title', data=top10_videos)
+print(ax1)
+print(video_data)
```

Comment:
This codebase is completely unacceptable, please look at how other classes are implemented. And follow that.